File: android/app/src/main/java/com/smilepile/onboarding/screens/WelcomeScreen.kt
package com.smilepile.onboarding.screens

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun WelcomeScreen(
    onGetStarted: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(horizontal = 24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.SpaceBetween
    ) {
        Spacer(modifier = Modifier.height(60.dp))

        // Logo and title
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Icon(
                imageVector = Icons.Default.PhotoLibrary,
                contentDescription = null,
                modifier = Modifier.size(100.dp),
                tint = Color(0xFFFF6B6B)
            )

            Text(
                text = "SmilePile",
                fontSize = 36.sp,
                fontWeight = FontWeight.Bold,
                color = MaterialTheme.colorScheme.primary
            )

            Text(
                text = "A safe and fun photo gallery for kids",
                fontSize = 18.sp,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(horizontal = 16.dp)
            )
        }

        // Features list
        Column(
            verticalArrangement = Arrangement.spacedBy(20.dp),
            modifier = Modifier.padding(vertical = 40.dp)
        ) {
            FeatureRow(
                icon = Icons.Default.Folder,
                title = "Organize photos",
                description = "Create colorful categories",
                tintColor = Color(0xFFFF6B6B)
            )

            FeatureRow(
                icon = Icons.Default.Photo,
                title = "Import memories",
                description = "Add your favorite photos",
                tintColor = Color(0xFF4ECDC4)
            )

            FeatureRow(
                icon = Icons.Default.Lock,
                title = "Stay secure",
                description = "Optional PIN protection",
                tintColor = Color(0xFF45B7D1)
            )
        }

        // Get Started button
        Button(
            onClick = onGetStarted,
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 32.dp)
                .height(56.dp),
            colors = ButtonDefaults.buttonColors(
                containerColor = Color(0xFFFF6B6B)
            ),
            shape = MaterialTheme.shapes.medium
        ) {
            Text(
                text = "Get Started",
                fontSize = 18.sp,
                fontWeight = FontWeight.Medium
            )
        }
    }
}

@Composable
private fun FeatureRow(
    icon: ImageVector,
    title: String,
    description: String,
    tintColor: Color
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.spacedBy(16.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            modifier = Modifier.size(32.dp),
            tint = tintColor
        )

        Column {
            Text(
                text = title,
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
                color = MaterialTheme.colorScheme.onSurface
            )

            Text(
                text = description,
                fontSize = 14.sp,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }
    }
}

File: ios/SmilePile/Onboarding/Screens/WelcomeScreen.swift
import SwiftUI

struct WelcomeScreen: View {
    @ObservedObject var coordinator: OnboardingCoordinator
    @Environment(\.typography) var typography: Typography
    @Environment(\.horizontalSizeClass) var sizeClass
    @State private var showPrivacyPolicy = false

    var body: some View {
        VStack(spacing: 20) {
            Spacer()

            // Logo and title section
            VStack(spacing: 12) {
                // SmilePile logo icon only (sharp rendering)
                Image("SmilePileLogo")
                    .resizable()
                    .renderingMode(.original)
                    .interpolation(.high)
                    .antialiased(true)
                    .aspectRatio(contentMode: .fit)
                    .frame(width: 100, height: 100)

                // Multicolor app name (separate component with shadow)
                MulticolorSmilePileLogo(fontSize: 36, showShadow: true)

                // Tagline
                Text("A safe and fun photo gallery for EVERYONE")
                    .font(typography.bodyLarge)
                    .foregroundColor(.secondary)
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, 40)
            }

            Spacer()

            // Features list
            Group {
                if sizeClass == .regular {
                    HStack {
                        Spacer()
                        VStack(alignment: .leading, spacing: 24) {
                            FeatureRow(
                                icon: "square.stack",
                                iconColor: .smilePileYellow,
                                title: "Organize photos into piles",
                                description: "Create colorful piles for your photos"
                            )

                            FeatureRow(
                                icon: "arrow.up.left.and.arrow.down.right",
                                iconColor: .smilePileOrange,
                                title: "Distraction-free mode",
                                description: "Good for kids (and everyone else)"
                            )

                            FeatureRow(
                                icon: "lock.fill",
                                iconColor: .smilePileGreen,
                                title: "Optional PIN protection",
                                description: "Prevent inadvertent changes"
                            )
                        }
                        .frame(maxWidth: 600)
                        Spacer()
                    }
                    .padding(.horizontal, 40)
                } else {
                    VStack(alignment: .leading, spacing: 24) {
                        FeatureRow(
                            icon: "square.stack",
                            iconColor: .smilePileYellow,
                            title: "Organize photos into piles",
                            description: "Create colorful piles for your photos"
                        )

                        FeatureRow(
                            icon: "arrow.up.left.and.arrow.down.right",
                            iconColor: .smilePileOrange,
                            title: "Distraction-free mode",
                            description: "Good for kids (and everyone else)"
                        )

                        FeatureRow(
                            icon: "lock.fill",
                            iconColor: .smilePileGreen,
                            title: "Optional PIN protection",
                            description: "Prevent inadvertent changes"
                        )
                    }
                    .padding(.horizontal, 40)
                }
            }

            Spacer()

            // Buttons
            VStack(spacing: 12) {
                // Get Started button (primary action)
                Button(action: {
                    coordinator.navigateToNext()
                }) {
                    Text("Start Fresh")
                        .font(typography.bodyLarge)
                        .fontWeight(.bold)
                        .foregroundColor(.white)
                        .frame(maxWidth: 400)
                        .frame(height: 56)
                        .frame(maxWidth: .infinity)
                        .background(Color.smilePileBlue)
                        .cornerRadius(12)
                }

                // Try Demo button (secondary action)
                Button(action: {
                    coordinator.enterDemoMode()
                }) {
                    HStack(spacing: 6) {
                        Image(systemName: "star.fill")
                            .font(.system(size: 14))
                        Text("Try Demo")
                            .font(typography.bodyMedium)
                            .fontWeight(.medium)
                    }
                    .foregroundColor(.smilePileBlue)
                    .frame(maxWidth: 400)
                    .frame(height: 48)
                    .frame(maxWidth: .infinity)
                }
            }
            .padding(.horizontal, 40)

            // Privacy Policy Link
            Button(action: {
                showPrivacyPolicy = true
            }) {
                Text("Privacy Policy")
                    .font(typography.bodySmall)
                    .foregroundColor(.secondary)
                    .underline()
            }
            .padding(.top, 8)
            .padding(.bottom, 50)
        }
        .sheet(isPresented: $showPrivacyPolicy) {
            PrivacyPolicyView()
        }
    }
}

struct FeatureRow: View {
    let icon: String
    let iconColor: Color
    let title: String
    let description: String
    @Environment(\.typography) var typography: Typography

    var body: some View {
        HStack(alignment: .top, spacing: 16) {
            Image(systemName: icon)
                .font(.title2)
                .foregroundColor(iconColor)
                .frame(width: 30)

            VStack(alignment: .leading, spacing: 4) {
                Text(title)
                    .font(typography.bodyMedium)
                    .fontWeight(.semibold)
                    .foregroundColor(.primary)

                Text(description)
                    .font(typography.bodySmall)
                    .foregroundColor(.secondary)
            }
        }
    }
}